### DISCUSSION TITLE
Implicitly scoped using statement without explicit assignment

### DISCUSSION THREAD

**Discussion by r30e** [dotnet/csharplang#8214] [Language Ideas] [answered]
The introduction of implicitly scoped suing statments have been great for C# and have improved readbility greatly. However, they have one limitation vs traditional explicitly-scoped using statements: They require assigning a variable. There are many situations in which I want to `use` an `IDisposable` without actually assigning a variable. All of these situations require a compromise on readability - either you have to explicitly scope the using statement or you have to assigned an unused variable.

logging:
```csharp
using(logger.BeginScope(myState))
{
  // ...
}
```

FluentAssertions:
```csharp
using(new AssertionScope())
{
  // ...
}
```

method returning IDisposable and `out var`:
```csharp
using(RentBuffer(out Memory<T> memory))
{
  // ...
}
```

## Proposal
Allow implicitly-scoped using statements to exist without a(n explicit) variable assignment:

logging:
```csharp
using logger.BeginScope(myState);
// ...
```

FluentAssertions:
```csharp
using new AssertionScope();
// ...
```

IDidposable & `out var`:
```csharp
using RentBuffer(out Memory<T> memory);
// ...
```

This would allow this situations to benefit from the increased readability of the implict-scoping.

**Comment by ufcpp** [comment, marked as answer]
https://github.com/dotnet/csharplang/discussions/7882

**Reply by r30e** [reply]
Thanks, I did search but couldn't find this. Have closed as duplicate
